tests/test_apply.py

```diff
@@ -934,6 +962,24 @@ def circuit():
             ("ControlledPhaseShift", [-math.pi], [-1 / 2, -1 / 2]),
             ("ControlledPhaseShift", [math.pi / 2], [-1 / 2, -1 / 2]),
             ("ControlledPhaseShift", [math.pi], [-1 / 2, -1 / 2]),
+            ("IsingXX", [math.pi], [1 / 2, 1 / 2]),
+            ("IsingYY", [math.pi], [1 / 2, 1 / 2]),
+            ("IsingZZ", [math.pi], [-1 / 2, -1 / 2]),
+            ("IsingXX", [0], [-1 / 2, -1 / 2]),
+            ("IsingYY", [0], [-1 / 2, -1 / 2]),
+            ("IsingZZ", [0], [-1 / 2, -1 / 2]),
+            ("IsingXX", [math.pi / 2], [0, 0]),
+            ("IsingYY", [math.pi / 2], [0, 0]),
+            ("IsingZZ", [math.pi / 2], [-1 / 2, -1 / 2]),
+            ("SingleExcitation", [math.pi], [-1 / 2, -1 / 2]),
+            ("SingleExcitation", [0], [-1 / 2, -1 / 2]),
+            ("SingleExcitation", [math.pi / 2], [-1 / 2, -1 / 2]),
+            ("SingleExcitationMinus", [math.pi], [-1 / 2, -1 / 2]),
+            ("SingleExcitationMinus", [0], [-1 / 2, -1 / 2]),
+            ("SingleExcitationMinus", [math.pi / 2], [-1 / 2, -1 / 2]),
+            ("SingleExcitationPlus", [math.pi], [-1 / 2, -1 / 2]),
+            ("SingleExcitationPlus", [0], [-1 / 2, -1 / 2]),
+            ("SingleExcitationPlus", [math.pi / 2], [-1 / 2, -1 / 2]),
```

Comment:
I always worry about tests with 0, pi, pi/2 as params. I wonder if it makes sense to use your favourite reals here instead (though I know that is more painful to code-up).